pfcli/service/client/checkpoint.py

```diff
@@ -211,7 +211,8 @@ def upload_files(
         spu_urls = [url_info["upload_url"] for url_info in spu_url_dicts]
 
         with tqdm(total=total_size, unit="B", unit_scale=True, unit_divisor=1024) as t:
-            with ThreadPoolExecutor() as executor:
+            # NOTE: excessive concurrency may results in "No buffer space available" error.
+            with ThreadPoolExecutor(max_workers=max_workers // 2) as executor:
```

Comment:
Does dividing by 2 always ok (no buffer space error)? when there are too many files?

Reply:
I am testing uploading 40 different files each with 1GB size, but I do not see any difference in uploading throughput when varying `max_workers`.
Could you elaborate on your testing scenario?

Reply:
Considering the comment of @ktaebum, I made a change to use fixed-size threads for uploading files. Please see https://github.com/friendliai/periflow-cli/pull/104#issuecomment-1351753309.